core/Archive/DataCollection.php

```diff
@@ -147,11 +147,17 @@ public function &get($idSite, $period)
      * @param string $period eg, '2012-01-01,2012-01-31'
      * @param string $name eg 'nb_visits'
      * @param string $value eg 5
+     * @param array  $meta Optional metadata to add to the row
      */
-    public function set($idSite, $period, $name, $value)
+    public function set($idSite, $period, $name, $value, $meta = null)
```

Comment:
we actually can use type hints a bit more to ensure the correct type is given to a method
```suggestion
    public function set($idSite, $period, $name, $value, array $meta = null)
```

We could actually do that for every new parameter or method we introduce, so at some point we don't need all those additional type checks anymore. Feel free to go through the other code in this PR as well and add some type hints where useful.

Reply:
Will do, I'm a big fan of type hints but haven't been using them in an effort to be consistent with existing code.